File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    dic = {}
    for runner in participant:
        if runner not in dic:
            dic[runner] = 1
        else:
            dic[runner] += 1
            
    for finisher in completion:
        if dic[finisher] == 1:
            del dic[finisher]
        elif dic[finisher] > 1:
            dic[finisher] -= 1
    
    return list(dic.keys())[0]
    

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> map = new HashMap<>();
        for (String s : participant) {
            map.put(s, map.getOrDefault(s, 0) + 1);
        }
        
        for (String s : completion) {
            Integer n = map.get(s);
            if (n-1 == 0) {
                map.remove(s);
            } else {
                map.put(s, n-1);
            }
        }
        
        String answer = "";
        for (String key : map.keySet()) {
            answer = key;
        }
        
        return answer;
    }
}